### PR TITLE
CommunicationPreferences settings page

### PR DESCRIPTION
**Pull request by joshblum**
screen shot 2016-04-28 at 12 32 24

### CODE REVIEW

**Comment by marcua** [comment]
UX: description of each communication type

**Comment by marcua** [comment]
UX: no bullet point

**Comment by joshblum** [comment]
screen shot 2016-04-28 at 17 07 56

**Comment by joshblum** [comment]
@marcua where do you suggest to store the descriptions? The should we put it in the model with the `ChoicesEnum`? is there a way for the enum to support tuples?

File: orchestra/accounts/bitformfield.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Copy pasted from
+# https://raw.githubusercontent.com/greyside/django-bitfield/04efae91d292bffb569bd7cc13e5c941e42e8698/bitfield/forms.py
+from django.forms import CheckboxSelectMultiple, IntegerField, ValidationError
```
Comment (kkamalov) [inline orchestra/accounts/bitformfield.py:3]: make it three lines of imports

File: orchestra/accounts/tests/test_views.py [hunk 2]
```diff
@@ -63,3 +65,54 @@ def test_missing_fields(self):
...
+    def test_disable_email(self):
+
+        # email is unset and therefore false
+        data = self._get_mock_data()
+
+        response = self.request_client.post(self.url, data)
+        self.assertTrue(response.context['success'])
+
+        self.comm_pref.refresh_from_db()
+        self.assertEqual(self.comm_pref.methods.email,
```
Comment (kkamalov) [inline orchestra/accounts/tests/test_views.py]: `self.assertFalse(self.comm_pref.methods.email.is_set)`

File: orchestra/accounts/views.py [hunk 3]
```diff
@@ -1,16 +1,20 @@
 from django.conf import settings
+from orchestra.accounts.bitformfield import BitFormField
```
Comment (kkamalov) [inline orchestra/accounts/views.py]: can you put import a little bit below. looks a bit off place

File: orchestra/accounts/views.py [hunk 4]
```diff
@@ -99,3 +109,56 @@ def post(self, request, *args, **kwargs):
...
+            'comm_pref_formset': comm_pref_formset,
+        })
+
+    def post(self, request, *args, **kwargs):
+        comm_pref_formset = self.CommunicationPreferenceFormSet(
+            data=request.POST)
+        self.set_method_choices(comm_pref_formset)
+
+        success = comm_pref_formset.is_valid()
+        if success:
```
Comment (kkamalov) [inline orchestra/accounts/views.py]: `if comm_pref_formset.is_valid()`

File: orchestra/accounts/bitformfield.py (hunk 1 above)
Comment (joshblum) [inline orchestra/accounts/bitformfield.py:3]: this file is copied so we should just leave as is

File: orchestra/accounts/tests/test_views.py (hunk 2 above)
Comment (joshblum) [inline orchestra/accounts/tests/test_views.py]: done :)

File: orchestra/accounts/views.py (hunk 3 above)
Comment (joshblum) [inline orchestra/accounts/views.py]: done

**Comment by marcua** [comment]
This is something like what @meredithblumenstock suggested.  She mentioned making the active tab white, but I think it's OK to ask her about this in the morning and merge it sooner after you do these changes.

img_20160428_215912568

**Comment by marcua** [comment]
Description text can be:

`When a task status changes (e.g., moved from in review to returned from reviewer), you will automatically be alerted in a project slack group.  You can optionally receive email alerts.`

**Comment by marcua** [comment]
lgtm aside from the UX changes

**Comment by joshblum** [comment]
screen shot 2016-04-28 at 22 47 11

will discuss further UX changes with @meredithblumenstock

**Pull request merged**